verilog/preprocessor/verilog_preprocess.cc

```diff
@@ -552,6 +554,115 @@ absl::Status VerilogPreprocess::HandleEndif(
   return absl::OkStatus();
 }
 
+// Handle `include directives.
+absl::Status VerilogPreprocess::HandleInclude(
+    TokenStreamView::const_iterator iter,
+    const StreamIteratorGenerator& generator) {
+  // TODO(karimtera): how to differentiate between <file> and "file"?? both are
+  // the same token, need to edit the lexer.
+  TokenStreamView::const_iterator token_iter = generator();
+  auto file_token_iter = *token_iter;
+  if (file_token_iter->token_enum() != TK_StringLiteral) {
+    preprocess_data_.errors.push_back(
+        {**token_iter, "Expected a path to a SV file."});
+    return absl::InvalidArgumentError("Expected a path to a SV file.");
+  }
+  // Currently the file path looks like "path", we need to remove "".
+  const auto& token_text = file_token_iter->text();
+
+  std::filesystem::path file_path =
+      std::string(token_text.substr(1, token_text.size() - 2));
+
+  std::string file_path_to_search = file_path.string();
+
```

Comment:
Opening the include file looks like it is similar to what `VerilogProject` provides in `VerilogProject::OpenIncludedFile()`.

Might be worthwhile documenting the differences here and why we need a separate implementation (or maybe even use some of it directly if you weren't aware of it yet). Documentation then helps later potential unification efforts.

Reply:
I added my thoughts in the description of this PR.

Reply:
It is best to add it to the declaration or implementation comment, as the PR description can't be easily be read just looking at the code, so the discussion would be lost.